test/org/loklak/client/SuggestClientTest.java

```diff
@@ -0,0 +1,132 @@
+    private final String QUERY_KEY_QUERY_LENGTH = "query_length";
+    private final String QUERY_KEY_TIMEZONE_OFFSET = "timezoneOffset";
+    private final String QUERY_KEY_RET_NEXT = "retrieval_next";
+    private final String QUERY_KEY_SCORE_RET = "score_retrieval";
+    private final String QUERY_KEY_QUERY_LAST = "query_last";
+    private final String QUERY_KEY_EXPECTED_NEXT = "expected_next";
+    private final String QUERY_KEY_SCORE_SUGGEST = "score_suggest";
+    private final String QUERY_KEY_RET_COUNT = "retrieval_count";
+    private final String QUERY_KEY_QUERY_FIRST = "query_first";
+
+    private static List<String> sSuggestApiParamValues = new ArrayList<>();
+    private static String sSuggestApiUrl;
+
+    /**
+     * Creates the required url to send API request.
+     */
+    @BeforeClass
+    public static void createSuggestApiUrl() {
+        try {
+            Field[] fields = SUGGEST_CLIENT_CLASS.getDeclaredFields();
+            for (Field field : fields) {
+                String fieldName = field.getName();
+                for (String param : SUGGEST_API_PARAMS) {
+                    if (fieldName.equals(param)) {
+                        field.setAccessible(true);
+                        sSuggestApiParamValues.add(field.get(fieldName).toString());
+                    }
+                }
+            }
+        } catch (IllegalAccessException e) {
+            fail("Suggest API field doesn't exist.");
+        }
+        sSuggestApiUrl = HOST_SERVER_URL
+                + sSuggestApiParamValues.get(0) + SEARCH_QUERY // param: base API url and query
+                + sSuggestApiParamValues.get(1) + sSuggestApiParamValues.get(2); // param: minified
+    }
+
+    /**
+     * Tests by checking whether the returned JSONObject has required keys.
+     */
+    @Test
+    public void testSuggest() {
+        JSONObject jsonObject = null;
+        try {
+            jsonObject = JsonIO.loadJson(sSuggestApiUrl);
+        } catch (IOException | JSONException e) {
+            if (e instanceof JSONException) {
+                fail("JSON parsing unsuccessful.");
+            }
+            return;
+        }
+
+        assertTrue(jsonObject.has(JSON_KEY_SEARCH_METADATA));
+        assertTrue(jsonObject.has(JSON_KEY_QUERIES));
+
+        JSONObject jsonMetadataObject = jsonObject.getJSONObject(JSON_KEY_SEARCH_METADATA);
+        assertTrue(jsonMetadataObject.has(SM_KEY_COUNT));
+        assertTrue(jsonMetadataObject.has(SM_KEY_HITS));
+        assertTrue(jsonMetadataObject.has(SM_KEY_QUERY));
+        assertTrue(jsonMetadataObject.has(SM_KEY_ORDER));
+        assertTrue(jsonMetadataObject.has(SM_KEY_ORDER_BY));
+        assertTrue(jsonMetadataObject.has(SM_KEY_CLIENT));
+
+        JSONArray jsonQueryArray = jsonObject.getJSONArray(JSON_KEY_QUERIES);
+        if (jsonQueryArray.length() > 0) {
+            JSONObject queryObject = jsonQueryArray.getJSONObject(0);
+            assertTrue(queryObject.has(QUERY_KEY_QUERY));
```

Comment:
try using assertThat 
see this -> https://objectpartners.com/2013/09/18/the-benefits-of-using-assertthat-over-other-assert-methods-in-unit-tests/